src/ledger/LedgerManagerImpl.cpp

```diff
@@ -1659,8 +1684,9 @@ LedgerManagerImpl::setLastClosedLedger(
     // bucket state, there's no tx-apply state to snapshot, in this one
     // case we will prime the tx-apply-state's soroban module cache using
     // a snapshot _from_ the LCL state.
-    mApplyState.compileAllContractsInLedger(
-        mLastClosedLedgerState->getBucketSnapshot(), lv);
+    auto const& snapshot = mLastClosedLedgerState->getBucketSnapshot();
+    mApplyState.compileAllContractsInLedger(snapshot, lv);
+    mApplyState.populateInMemorySorobanState(snapshot);
```

Comment:
This is another place where this cache will get out-of-date before p23 kicks in.

Reply:
This is fine. The cache will just be empty since no soroban state exists, but it won't be "out of date" since we apply all Soroban entries to the cache starting at p20, which is the first point at which such entries can exist.